src/management-system-v2/lib/useColumnWidth.tsx

```diff
@@ -1,206 +1,255 @@
+      }),
+    };
+  }) as TableColumnsType<any>;
+
+  return useTruncateColumnText(columsWithResize);
+  // return columsWithResize;
+};
 
-        /* Reset */
-        resizingColumn.current = '';
-        startX.current = 0;
+export const ResizableTitle = (props: any) => {
+  const { onResize, width, ...restProps } = props;
+
+  if (!width) {
+    return <th {...restProps} />;
+  }
+
+  return (
+    <Resizable
+      className={styles['react-resizable']}
+      width={width}
+      height={0}
+      handle={
+        <span
+          className={styles['react-resizable-handle']}
+          onClick={(e) => {
+            e.stopPropagation();
+          }}
+          // style={{ backgroundColor: 'red' }} /* Uncomment to see Handles */
+        >
+          {/* {width} */}
+        </span>
       }
-    },
-    [columnWidths],
+      onResize={onResize}
+      draggableOpts={{ enableUserSelectHack: false }}
+    >
+      <th {...restProps} />
+    </Resizable>
   );
+};
+
+export const useTruncateColumnText = (columns: NonNullable<TableProps['columns']>) => {
+  const truncatedColumns = useMemo(() => {
+    return columns.map((column: any) => {
+      return {
+        ...column,
+        render: (text: any, record: any, rowIndex: number) => {
+          const fallBackText = text.highlighted
+            ? text.highlighted
+            : text; /* In case fuzzy-search is used */
+          const newRender = column.render
+            ? () => column.render(text, record, rowIndex)
+            : () => fallBackText;
+          return (
+            <>
+              <TruncatedCell width={column.width} innerRender={newRender}></TruncatedCell>
+            </>
+          );
+        },
+      };
+    });
+  }, [columns]);
+
+  return truncatedColumns;
+};
+
+type TruncateType = {
+  width: number | string;
+  innerRender: () => JSX.Element | string;
+};
+
+const TruncatedCell: FC<TruncateType> = ({ width, innerRender }) => {
+  const [overFlowing, setOverFlowing] = useState(false);
+  const containerRef = useRef<HTMLDivElement>(null);
 
   useEffect(() => {
-    if (!document) return;
-
-    // @ts-ignore
-    document.addEventListener('mouseup', handleMouseUp);
-    // @ts-ignore
-    document.addEventListener('mousemove', handleMouseMove);
-
-    return () => {
-      // @ts-ignore
-      document.removeEventListener('mouseup', handleMouseUp);
-      // @ts-ignore
-      document.removeEventListener('mousemove', handleMouseMove);
-    };
-  }, [handleMouseUp, handleMouseDown]);
+    if (!containerRef.current || typeof width !== 'number') return;
 
-  return columns.map((column: any) => {
-    if (notResizeabel.includes(column.key)) return column;
-    return {
-      ...column,
-      title: (
-        <div style={{ position: 'relative' }} className="ColumnWidthParent">
-          {column.title as ReactNode}
+    const widths = getWidthsOfInnerElements(containerRef.current);
+
+    if (widths.some((w) => w > width)) {
```

Comment:
Mazbe this should be adapted in the future, as the total width can be determined by two direct childs, that are displayed next to each other (such as the process / folder icon and the title)

Reply:
yeah, would be good

Reply:
Changed it to also check for elements that are displayed inline (i.e. use the sum of their width)